### PR TITLE
Added ARM And Bicep Template for Azure synapse

### PR DESCRIPTION
**Pull request by dassayantan1**
ARM and Bicep template for Azure synapse workspace.
While creation of the synapse resource we will also create diagnostic settings for that particular synapse resource (Azure_SynapseWorkspace_Audit_Enable_DiagnosticSettings_Trial).

### CODE REVIEW

File: 08-Secure by default/Secure Templates/ARM/Synapse.json [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        "sqlAdminUser":{
+            "type": "string",
+            "metadata":{
+                "description": "sql admin user "
+            }
+        },
+        "sqlAdminUserPassword":{
+            "type": "string",
+            "metadata":{
+                "description": "Password for sql admin user."
```
Comment (TarunKrShukla) [inline 08-Secure by default/Secure Templates/ARM/Synapse.json:32]: NIT: 
1. Either remove or keep full stop (.) in every description.
2. Start description with uppercase  at everyplace.

File: 08-Secure by default/Secure Templates/ARM/Synapse.json [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+            "location": "[resourceGroup().location]",
+            "identity": {
+                "type": "SystemAssigned"
+            },
+            "properties": {
+                "defaultDataLakeStorage": {
+                    "accountUrl": "[parameters('dataLakeStorageAccountUrl')]",
+                    "filesystem": "[parameters('dataLakeStorageFileSystem')]"
+                },
+                "sqlAdministratorLogin": "[parameters('sqlAdminUser')]",
```
Comment (TarunKrShukla) [inline 08-Secure by default/Secure Templates/ARM/Synapse.json:58]: Please check if directly we can use MS Entra ID authentication here.

File: 08-Secure by default/Secure Templates/ARM/Synapse.json [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+                "sqlAdministratorLoginPassword": "[parameters('sqlAdminUserPassword')]"
+            }
+        },
+        {
+            // This one is required for diagnostic settings. (Azure_SynapseWorkspace_Audit_Enable_DiagnosticSettings_Trial)
```
Comment (TarunKrShukla) [inline 08-Secure by default/Secure Templates/ARM/Synapse.json:63]: NIT: Please remove text: This one is required for diagnostic settings

Just keep control Id and display name like / Azure_ServiceFabric_DP_Set_Property_ClusterProtectionLevel - The ClusterProtectionLevel property must be set to "EncryptAndSign".

File: 08-Secure by default/Secure Templates/ARM/Synapse.json (hunk 1 above)
Comment (TarunKrShukla) [reply 08-Secure by default/Secure Templates/ARM/Synapse.json:32]: Do same in bicep template as well.

File: 08-Secure by default/Secure Templates/Bicep/Synapse.bicep [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  location: resourceGroup().location
+  identity: {
+    type: 'SystemAssigned'
+  }
+  properties: {
+    defaultDataLakeStorage: {
+      accountUrl: dataLakeStorageAccountUrl
+      filesystem: dataLakeStorageFileSystem
+    }
+    sqlAdministratorLogin: sqlAdminUser
```
Comment (TarunKrShukla) [inline 08-Secure by default/Secure Templates/Bicep/Synapse.bicep:30]: Same as ARM template.

File: 08-Secure by default/Secure Templates/ARM/Synapse.json (hunk 1 above)
Comment (dassayantan1) [reply 08-Secure by default/Secure Templates/ARM/Synapse.json:32]: Fixed.

File: 08-Secure by default/Secure Templates/ARM/Synapse.json (hunk 3 above)
Comment (dassayantan1) [reply 08-Secure by default/Secure Templates/ARM/Synapse.json]: Fixed.

File: 08-Secure by default/Secure Templates/Bicep/Synapse.bicep (hunk 4 above)
Comment (dassayantan1) [reply 08-Secure by default/Secure Templates/Bicep/Synapse.bicep:30]: Made changes here as well.

File: 08-Secure by default/Secure Templates/ARM/Synapse.json [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+                "description": "Storage account id to which the diagnostic logs and metrices will be sent."
+            }
+        }
+    },
+    "functions": [],
+    "variables": {},
+    "resources": [
+        {
+            "type": "Microsoft.Synapse/workspaces",
+            "apiVersion": "2019-06-01-preview",
```
Comment (jaiswalricha) [inline 08-Secure by default/Secure Templates/ARM/Synapse.json:47]: we don't have stable version to check the same?

Comment (dassayantan1) [reply 08-Secure by default/Secure Templates/ARM/Synapse.json]: Changed to 2021-06-01

**Review by TarunKrShukla — APPROVED** [review]

**Pull request merged**